androidApp/src/main/java/com/mbta/tid/mbta_app/android/pages/NearbyTransitPage.kt

```diff
@@ -216,6 +218,8 @@ fun NearbyTransitPage(
                 }
             }
             composable<SheetRoutes.NearbyTransit> {
+                // for ViewModel reasons, must be within the `composable` to be the same instance
+                val nearbyViewModel: NearbyTransitViewModel = koinViewModel()
```

Comment:
Originally, I had `nearbyViewModel: NearbyTransitViewModel = koinViewModel()` as an argument to `NearbyTransitPage`, but since navigation states [can own ViewModels](https://developer.android.com/topic/libraries/architecture/viewmodel#lifecycle), calling `koinViewModel()` outside this `composable` creates a different instance than the one that's created as an argument in `NearbyTransitView`, but calling it here gives the same instance.

Reply:
That is tricky. To clarify my own understanding, that would be an issue even if we weren't using koin and were instantiating the view models directly, is that right? I haven't been using koinViewModel regularly, and wondering about differences / benefits we get with it aside from test defaults.

Reply:
I believe that this is inherent to the Android ViewModel subsystem and not specific to Koin's integration with it. My understanding is that using `koinViewModel` lets us pass repositories via Koin into the ViewModel without having to define a custom `ViewModelProvider.Factory` or deal with `CreationExtras`.